google/cloud/storage/internal/async/reader_connection_factory.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/async/reader_connection_factory.h"
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+void UpdateGeneration(google::storage::v2::ReadObjectRequest& request,
+                      storage::Generation generation) {
+  if (request.generation() != 0 || !generation.has_value()) return;
+  request.set_generation(generation.value_or(0));
```

Comment:
Why `generation.value_or()` when we know `generation.has_value()`?

Reply:
Fixed.